bundles/org.openhab.binding.mecmeter/src/main/java/org/openhab/binding/mecmeter/handler/MecMeterHandler.java

```diff
@@ -0,0 +1,408 @@
+                    return new QuantityType<>(response.getReverseActiveHarmonicEnergyPhase1(), Units.KILOWATT_HOUR);
+                case MecMeterBindingConstants.REVERSE_ACTIVE_HARM_ENERGY_PHASE_2:
+                    return new QuantityType<>(response.getReverseActiveHarmonicEnergyPhase2(), Units.KILOWATT_HOUR);
+                case MecMeterBindingConstants.REVERSE_ACTIVE_HARM_ENERGY_PHASE_3:
+                    return new QuantityType<>(response.getReverseActiveHarmonicEnergyPhase3(), Units.KILOWATT_HOUR);
+                case MecMeterBindingConstants.REVERSE_ACTIVE_HARM_ENERGY_ALL:
+                    return new QuantityType<>(response.getReverseActiveHarmonicEnergySum(), Units.KILOWATT_HOUR);
+                case MecMeterBindingConstants.REVERSE_REACTIVE_ENERGY_PHASE_1:
+                    return new QuantityType<>(response.getReverseReactiveEnergyPhase1(), Units.VAR_HOUR);
+                case MecMeterBindingConstants.REVERSE_REACTIVE_ENERGY_PHASE_2:
+                    return new QuantityType<>(response.getReverseReactiveEnergyPhase2(), Units.VAR_HOUR);
+                case MecMeterBindingConstants.REVERSE_REACTIVE_ENERGY_PHASE_3:
+                    return new QuantityType<>(response.getReverseReactiveEnergyPhase3(), Units.VAR_HOUR);
+                case MecMeterBindingConstants.REVERSE_REACTIVE_ENERGY_ALL:
+                    return new QuantityType<>(response.getReverseReactiveEnergySum(), Units.VAR_HOUR);
+
+                /* Apparent Energy */
+                case MecMeterBindingConstants.APP_ENERGY_PHASE_1:
+                    return new QuantityType<>(response.getApparentEnergyConsumptionPhase1(), Units.VOLT_AMPERE_HOUR);
+                case MecMeterBindingConstants.APP_ENERGY_PHASE_2:
+                    return new QuantityType<>(response.getApparentEnergyConsumptionPhase2(), Units.VOLT_AMPERE_HOUR);
+                case MecMeterBindingConstants.APP_ENERGY_PHASE_3:
+                    return new QuantityType<>(response.getApparentEnergyConsumptionPhase3(), Units.VOLT_AMPERE_HOUR);
+                case MecMeterBindingConstants.APP_ENERGY_ALL:
+                    return new QuantityType<>(response.getApparentEnergyConsumptionSum(), Units.VOLT_AMPERE_HOUR);
+
+                /* Power Factor */
+                case MecMeterBindingConstants.POWER_FACTOR_PHASE_1:
+                    return new QuantityType<>(response.getPowerFactorPhase1(), Units.VOLT_AMPERE_HOUR);
+                case MecMeterBindingConstants.POWER_FACTOR_PHASE_2:
+                    return new QuantityType<>(response.getPowerFactorPhase2(), Units.VOLT_AMPERE_HOUR);
+                case MecMeterBindingConstants.POWER_FACTOR_PHASE_3:
+                    return new QuantityType<>(response.getPowerFactorPhase3(), Units.VOLT_AMPERE_HOUR);
+                case MecMeterBindingConstants.POWER_FACTOR_ALL:
+                    return new QuantityType<>(response.getPowerFactorSum(), Units.VOLT_AMPERE_HOUR);
```

Comment:
Are you sure these are Energy? The names sound dimensionless.

Reply:
Good point. It was in the original code with a unit, but I agree with you that it does not make sense. Have changed it to dimensionless.